src/gen-stochastic.js

```diff
@@ -406,4 +406,92 @@ class MarkovChain {
 		return c;
 	}
 }
-exports.MarkovChain = MarkovChain;
\ No newline at end of file
+
+class DeepMarkovChain {
+	constructor(data){
+		// transition probabilities table
+		this._table = new Map();
+		// train if dataset is provided
+		if (data) { this.train(data) };
+		// current state of markov chain
+		this._state = '';
+		// scoped random number generator
+		this.rng = seedrandom();
+	}
+	get table(){
+		// return copy of object
+		return this._table
+	}
+	clear(){
+		// empty the transition probabilities
+		this._table = new Map();
+	}
+	train(a, win){
```

Comment:
Maybe `train(a, win=2)` as a default value for win, because now using just `train([1, 2, 3, 4, 5])` will not train anything

Reply:
Definitely needs a default. Good catch.